src/topology/metric_space/basic.lean

```diff
@@ -2881,3 +2881,41 @@ instance metric_space_quot {α : Type u} [pseudo_metric_space α] :
     λxc yc zc, quotient.induction_on₃ xc yc zc (λx y z, pseudo_metric_space.dist_triangle _ _ _) }
 
 end eq_rel
+
+/-! ### `additive`, `multiplicative` -/
+
+open additive multiplicative
+
+section
+variables [has_dist X]
+
+instance : has_dist (additive X) := ‹has_dist X›
+instance : has_dist (multiplicative X) := ‹has_dist X›
+
+@[simp] lemma dist_of_mul (a b : X) : dist (of_mul a) (of_mul b) = dist a b := rfl
+@[simp] lemma dist_of_add (a b : X) : dist (of_add a) (of_add b) = dist a b := rfl
+@[simp] lemma dist_to_mul (a b) : dist (to_mul a : X) (to_mul b) = dist a b := rfl
+@[simp] lemma dist_to_add (a b) : dist (to_add a : X) (to_add b) = dist a b := rfl
+
+end
+
+section
+variables [has_nndist X]
+
+instance : has_nndist (additive X) := ‹has_nndist X›
+instance : has_nndist (multiplicative X) := ‹has_nndist X›
```

Comment:
These are probably a bad idea, all the other `nndist` instances go via `normed_group`

Reply:
Even though `additive` and `multiplicative` are type synonyms?

Reply:
Or is your point that no structure extends `has_nndist`, so I'm running the risk of diamonds? I just checked and there's no diamond here.

Reply:
> Even though `additive` and `multiplicative` are type synonyms? Do you also want me to let `has_edist` come from `pseudo_emetric_space` and `has_dist` come from `pseudo_metric_space`?

I think only `has_nndist` is relevant, as this is the only field for which definitional control is not provided in the derived classes. Yes, you can still ensure there's no diamond with the generated definition, but given that the isntance would be useless anyway I think it would be better to omit it to avoid that risk.

Note also that we don't have `prod.has_edist` or `prod.has_nndist`, so it's also reasonable to be consistent here